scripts/rabbitmq-k8s-runbook/rabbitmq-k8s-diagnostic-runbook

```diff
@@ -0,0 +1,456 @@
+    # Get logs for messaging topology operator
+    if [[ -n "${messaging_operator_image}" ]]; then
+        kubectl logs -l app.kubernetes.io/name=messaging-topology-operator --tail -1 -n ${RABBITMQ_OPERATORS_NAMESPACE} > $OUTPUT_DIRECTORY/operators/logs/messaging-operator.log
+        
+    fi
+
+    if [[ -n "${standby_operator_image}" ]]; then
+        kubectl logs -l app.kubernetes.io/name=standby-replication-operator --tail -1 -n ${RABBITMQ_OPERATORS_NAMESPACE} > $OUTPUT_DIRECTORY/operators/logs/standby-replication-operator.log
+        
+    fi
+
+    if [[ -n "${cert_manager_version}" ]]; then
+        kubectl logs -l app.kubernetes.io/instance=cert-manager --tail -1 -n cert-manager > $OUTPUT_DIRECTORY/operators/logs/cert-manager.log
+    fi
+
+    printf "\\n[INFO] Ending getting and storing operator pod logs\\n"
+
+}
+
+main() {
+    if [[ "$1" == "--help" ]]; then
+        usage
+        exit 0
+    fi
+
+    case "$1" in
+    # Get useful info and logs about the K8s cluster
+    "get_k8s_cluster_info")
+        shift 1
+        if [[ "$#" != "1" ]]; then
+            usage
+            exit 1
+        fi
+     
+        data_diagnostic_cluster_info $1
+        ;;
+    # Get info and logs on specific carvel commercial components: tanzu-essential, kapp, secret-gen controller, PackageRepository and PackageInstall
+    "get_carvel_components_info")
+        shift 1
+        if [[ "$#" -gt 3 || "$#" == 2 ]]; then
+            usage
+            exit 1
+        fi
+        if [[ "$#" -lt 3 ]]; then
+            namespace="default"
+            output_dir=$1
+        else
+            namespace=$2
+            output_dir=$3
+        fi
+
+        data_diagnostic_carvel_components_info $namespace $output_dir
+        ;;
+    # Get info and logs on the deployed operators in the given namespace
+    "get_operators_info")
+        shift 1
+        if [[ "$#" -gt 3 || "$#" == 2 ]]; then
+            usage
+            exit 1
+        fi
+        if [[ "$#" -lt 3 ]]; then
+            namespace="rabbitmq-system"
+            output_dir=$1
+        else
+            namespace=$2
+            output_dir=$3
+        fi
+        data_diagnostic_operators_info $namespace $output_dir
+        ;;
+    # Get info and logs of the deployed RabbitMQ cluster deployed in the specified namespace
+    "get_rabbitmq_cluster_info")
+        shift 1
+        if [[ "$#" -gt 3 || "$#" == 2 ]]; then
+            usage
+            exit 1
+        fi
+        if [[ "$#" -lt 3 ]]; then
+            namespace="default"
+            output_dir=$1
+        else
+            namespace=$2
+            output_dir=$3
+        fi
+        data_diagnostic_rabbitmq_cluster_info $namespace $output_dir
+        ;;
+    "help")
+        usage
+        ;;
+    *)
+        usage
+        exit 1
+        ;;
+    esac
+}
+
+if [[ "$#" -ge 1 ]]; then
```

Comment:
Should this be 2? If I run without an output directory I just get:

```shell
$ ./rabbitmq-k8s-diagnostic-runbook get_rabbitmq_cluster_info
./rabbitmq-k8s-diagnostic-runbook: line 435: $1: unbound variable
```

where I feel the helptext would be valuable.

Reply:
Yes! Copied pasted from our plugin and forgot to adapt it :)